### ISSUE TITLE
Add the option of Generating docs in the main section 

### ISSUE THREAD

**Issue by AayushSaini101** [asyncapi/studio#1104]
Reference: https://editor.swagger.io/ the Generator code section is present on the main screen, where as the in asyncapi studio devleoper need to click on three dots then need to click Generate doc/code.

For the easy accessiblity and increase the developer experience the generate code/docs option must be present in the main tab.

<img width="1892" alt="image" src="https://github.com/asyncapi/studio/assets/60972989/2b084cab-a2e1-42b1-b07c-7a0d3d5687d5">
<img width="1892" alt="image" src="https://github.com/asyncapi/studio/assets/60972989/67e1dc79-717c-4198-abdb-e1f5ee0e720e">

**Comment by KhudaDad414** [comment]
Totally, I think we should create some icons for those options and then show the current options as a tooltip.

**Comment by KhudaDad414** [comment]
@AayushSaini101 not sure if we can come up with an icon for each one, but it is easier to create four icons for each category and show a dropdown of those relevent options.
1) *import icon*.
   a) import from file
   b) import from url
   c) import from base64
2) *save icon*:
   a) save as yaml
   b) save as JSON
3) *convert icon*:
  a) convert to json/yaml
  b) convert to another AsyncAPI version
4) *generate code/doc icon*

**Comment by abhavgoel** [comment]
@AayushSaini101 @KhudaDad414 should the the three dots be removed completely from the editor page?

**Comment by KhudaDad414** [comment]
@abhavgoel yep. After we show the options in categories then nothing will remain for the the overflow menu.

**Comment by ShashwatPS** [comment]
@KhudaDad414 You mean something like this ?

https://github.com/user-attachments/assets/bcfbbbc4-7314-493f-b8cb-fd382cf26fc8

If this is good enough, i can remove the borders which are there in the generate section and make a PR.

**Comment by AayushSaini101** [comment]
> [@KhudaDad414](https://github.com/KhudaDad414) You mean something like this ?
> 
>  Screen.Recording.2025-01-04.at.11.29.25.AM.mov 
> If this is good enough, i can remove the borders which are there in the generate section and make a PR.

@ShashwatPS feel free to create a PR on this if you want thanks

**Comment by AayushSaini101** [comment]
I will take this

**Comment by aeworxet** [comment]
### Bounty Issue's service comment

Text labels: `bounty/2025-Q2`, `bounty/medium`, `bounty/coding`
First assignment to regular contributors: [2025-03-21 00:00:00 UTC+12:00](https://time.lol/#2025-03-21T00:00:00+12:00)
End Of Life after: [2025-04-30 23:59:59 UTC-12:00](https://time.lol/#2025-04-30T23:59:59-12:00)

@asyncapi/bounty_team

##### The Bounty Program is not a Mentorship Program. The accepted level of Bounty Program Participants is **Middle**/**Senior**.
##### Regular contributors should explain in meaningful words how they are going to approach the resolution process when expressing a desire to work on this Bounty Issue.

**Comment by abhavgoel** [comment]
Hi @AayushSaini101, the original issue states that the "Generate code/docs", should be in the main bar (I assumed you meant the side bar if I am not wrong).

+ @KhudaDad414 commented that we can categorize all such options. Lets finalize an approach and I'll start working on this.

Current PR adds the Generate code/docs option in the side bar. Also clarify what you meant by the main bar here.

**Comment by Shurtu-gal** [comment]
@abhavgoel this is a bounty issue please wait until you are assigned before moving forward.

**Comment by abhavgoel** [comment]
Should the options also be categorized as shown in the recording?

**Comment by AayushSaini101** [comment]
> Should the options also be categorized as shown in the recording?

yes, take the reference of the screen recording, https://editor.swagger.io/ reference this will help to enhance to user experience on the studio

**Comment by aeworxet** [comment]
### Bounty Issue's Timeline
|Complexity Level|Assignment Date (by GitHub)|Start Date (by BP Rules)|End Date (by BP Rules)|Draft PR Submission|Final PR Merge Start|Final PR Merge End|
|----------------|---------------------------|------------------------|----------------------|-------------------|--------------------|------------------|
|Medium          |2025-03-21              |[2025-04-07](https://time.lol/#2025-04-07T00:00:00+12:00)              |**[2025-05-18](https://time.lol/#2025-05-18T23:59:59-12:00)**        |[2025-04-20](https://time.lol/#2025-04-20T23:59:59-12:00)         |[2025-05-04](https://time.lol/#2025-05-04T23:59:59-12:00)         |**[2025-05-18](https://time.lol/#2025-05-18T23:59:59-12:00)**        |

##### Please note that the dates given represent deadlines, not specific dates, so if the goal is reached sooner, it's better.
##### Keep in mind the [responsibility](https://github.com/asyncapi/community/pull/897/files#diff-25ecb20a61754c225d6511ca08d7e7c9a14b9ca5a93e89bd42331e51c9ebf26dR222) for violations of the [Timeline](https://github.com/asyncapi/community/pull/897/files#diff-25ecb20a61754c225d6511ca08d7e7c9a14b9ca5a93e89bd42331e51c9ebf26dR95).
#
Assignee: @abhavgoel (`githubID`: [72388884](https://api.github.com/user/72388884))

**Comment by abhavgoel** [comment]
Approach after code analysis - 

- I plan to make 4 files each for different dropdown category (Import, Generate etc.) and move the code in them, so that they wont clutter the main file.

**Comment by abhavgoel** [comment]
https://github.com/user-attachments/assets/c460d95c-7d68-4369-b37c-7917faf28b73

@AayushSaini101 @Shurtu-gal does this look good?

**Comment by abhavgoel** [comment]
Can I raise a PR now, or according to the start date in the above comment?

**Comment by aeworxet** [comment]
@abhavgoel 
> Can I raise a PR now, or according to the start date in the above comment?

Yes, the expedited procedure is welcomed.

**Comment by Shurtu-gal** [comment]
@abhavgoel no need to worry regarding generator of `server-api` not working for v3.0.0. Just go ahead and start working on adding the test suites. I will fix the api at the earliest.

**Comment by nishthamonga6** [comment]
I want to contribute in this project

**Comment by AayushSaini101** [comment]
closed via: https://github.com/asyncapi/studio/pull/1184 bounty issue completed @aeworxet

**Comment by aeworxet** [comment]
### Bounty Issue Is Completed 🎉

@abhavgoel (`githubID`: [72388884](https://api.github.com/user/72388884)), please go to the dedicated [AsyncAPI Bounty Program 2025 page on Open Collective](https://opencollective.com/asyncapi/projects/asyncapi-bounty-program) and submit an invoice for `USD 200.00` (button '`ACTIONS`', dropdown option '`Submit expense`') with the expense title `Bounty studio#1104`, tag `bounty`, and full URL of this Bounty Issue in the description.

After submitting the invoice, please post the link to it in this Bounty Issue as a separate comment.

**Comment by aeworxet** [comment]
## **After submitting the invoice, please post the link to it in this Bounty Issue as a separate comment.**

**Comment by abhavgoel** [comment]
https://opencollective.com/asyncapi/projects/asyncapi-bounty-program/expenses/245680

**Comment by aeworxet** [comment]
✅
The invoice https://opencollective.com/asyncapi/projects/asyncapi-bounty-program/expenses/245680 was submitted by the same Bounty Program Participant who completed the Bounty Issue `studio#1104`.
